database/state.go

```diff
@@ -3,266 +3,127 @@ package database
 import (
 	"math/big"
 	"sync"
-	"fmt"
 	"github.com/ethereum/go-ethereum/common"
-	"github.com/ethereum/go-ethereum/consensus/ethash"
 	"github.com/ethereum/go-ethereum/core"
-	"github.com/ethereum/go-ethereum/core/state"
-	"github.com/ethereum/go-ethereum/core/vm"
 	"github.com/ethereum/go-ethereum/eth"
-	"github.com/ethereum/go-ethereum/ethdb"
 	"github.com/ethereum/go-ethereum/params"
 
 	ethTypes "github.com/ethereum/go-ethereum/core/types"
-	
 	tmtAbciTypes "github.com/tendermint/tendermint/abci/types"
-	tmtCode "github.com/tendermint/tendermint/abci/example/code"
+	tmtLog "github.com/tendermint/tendermint/libs/log"
 )
 
 //----------------------------------------------------------------------
-// EthState manages concurrent access to the intermediate workState object
+// State manages concurrent access to the intermediate blockState object
 // The eth tx pool fires TxPreEvent in a go-routine,
 // and the miner subscribes to this in another go-routine and processes the tx onto
-// an intermediate ethState. We used to use `unsafe` to overwrite the miner, but this
-// didn't work because it didn't affect the already launched go-routines.
-// So instead we introduce the Pending API in a small commit in go-eth
-// so we don't even start the miner there, and instead manage the intermediate ethState from here.
-// In the same commit we also fire the TxPreEvent synchronously so the order is preserved,
+// an intermediate state. We used to use `unsafe` to overwrite the miner, but this
+// didn't bs because it didn't affect the already launched go-routines.
+// So instead we introduce the Pending API in a small persist in go-eth
+// so we don't even start the miner there, and instead manage the intermediate state from here.
+// In the same persist we also fire the TxPreEvent synchronously so the order is preserved,
 // instead of using a go-routine.
+//
+// TODO: Rewrite this description, it doesn't make sense and this "State" Struct is even more suspicious.
 
-type EthState struct {
+type State struct {
```

Comment:
This is still referring to `Ethereum State` which you removed the `Eth` meaning from it. Why?

Reply:
Because it's actually abci state not ethereum. It was confusing me for quite some time. Ethereum has its own, additional state.

Reply:
I find it confusing because it's the DB state not Ethereum one. Ethereum one has its own state.

Reply:
Then, we should name it DbState, what do you think ?

Reply:
It's in DB pkg, that's enough. It's like the new "abci" package has only "mempool_connection" without the abci.